src/main/java/com/michelin/ns4kafka/controllers/AkhqClaimProviderController.java

```diff
@@ -114,6 +111,125 @@ public AKHQClaimResponseV2 generateClaimV2(@Valid @Body AKHQClaimRequest request
                 .build();
     }
 
+    /**
+     * List AKHQ claims (v025 and higher)
+     *
+     * @param request The AKHQ request
+     * @return The AKHQ claims
+     */
+    @Post("/v3")
+    public AKHQClaimResponseV3 generateClaimV3(@Valid @Body AKHQClaimRequest request) {
+
+        final List<String> groups = Optional.ofNullable(request.getGroups()).orElse(new ArrayList<>());
+
+        if (groups.contains(config.getAdminGroup())) {
+            return AKHQClaimResponseV3.ofAdmin(config.getNewAdminRoles());
+        }
+
+        List<AccessControlEntry> relatedACL = getAllAclForGroups(groups);
+
+        // Add all public ACLs
+        relatedACL.addAll(accessControlEntryService.findAllPublicGrantedTo());
+
+        Map<String, AKHQClaimResponseV3.Group> bindings = new LinkedHashMap<>();
+
+        // Start by creating a map that store permissions by role/cluster
+        relatedACL.forEach(acl -> {
+            String escapedString = Pattern.quote(acl.getSpec().getResource());
+            String patternRegex;
+
+            // Build the pattern regex based on the pattern type and the resource
+            if (acl.getSpec().getResourcePatternType() == AccessControlEntry.ResourcePatternType.PREFIXED) {
+                patternRegex = String.format("^%s.*$", escapedString);
+            } else {
+                patternRegex = String.format("^%s$", escapedString);
+            }
+            // Build the cluster regex
+            String patternCluster = String.format("^%s$", acl.getMetadata().getCluster());
+
+            String role = config.getNewRoles().get(acl.getSpec().getResourceType());
+            String key = role + "-" + acl.getSpec().getResource();
+
+            // If we already have permissions for the role and cluster, add the pattern to the existing one
+            if (bindings.containsKey(key)) {
+                bindings.get(key).getClusters().add(patternCluster);
+            } else {
+                // Otherwise we add a new one
+                bindings.put(key, AKHQClaimResponseV3.Group.builder()
+                        .role(role)
+                        .patterns(new ArrayList<>() {{
+                            add(patternRegex);
+                        }})
+                        .clusters(new ArrayList<>() {{
+                            add(patternCluster);
+                        }})
+                        .build());
+            }
+        });
+
+        List<AKHQClaimResponseV3.Group> result = optimizeV3Claim(bindings);
+
+        // Add access to all the schemas
+        result.add(AKHQClaimResponseV3.Group.builder()
```

Comment:
We can set the same pattern for the "registry-read" role and the "topic-read" if we need. In AKHQ <= 0.24.0 everyone sees all the schema because there is no restriction so I reproduced the same behaviour for AKHQ > 0.24.0. But because now we can apply a filter on schema based on the permission I can apply the same pattern for registry and read